pkg/sensors/exec/exec.go

```diff
@@ -68,6 +71,19 @@ func msgToExecveKubeUnix(m *processapi.MsgExecveEvent, exec_id string, filename
 	return kube
 }
 
+func msgToExecveAccountUnix(m *exec.MsgExecveEventUnix) {
+	mode := cgroups.GetDeploymentMode()
+	if ns, err := namespace.GetMsgNamespaces(m.Unix.Msg.Namespaces); err == nil {
```

Comment:
@tixxdz, thanks for comments. Maybe I misunderstand something. But look at the case: 
Tetragon is running on host. I ran a docker container. I this container I created a user with uid 1001. Than just called `whoami`.

```
"process_exec": {
    "process": {
      "exec_id": "YW5mZWRvdG9mZi1uaXg6MTEwNzA5MDEyMjAyMjc2MjoyOTU2ODM=",
      "pid": 295683,
      "uid": 1001,
      "cwd": "/",
      "binary": "/usr/bin/whoami",
      "flags": "execve rootcwd clone",
      "start_time": "2024-05-08T14:13:12.359738563Z",
      "auid": 4294967295,
      "docker": "ead4b91477a0d8affbde3db2317d89b",
      "parent_exec_id": "YW5mZWRvdG9mZi1uaXg6NDQ3NDYxOTUzMDAwMDAwMDoyOTQ1OTA=",
      "tid": 295683
    },
```
I think we don't want to resolve usernames for processes that are **not from host NS**.

This is `/etc/passwd` file from container:

```
nginx:x:101:101:nginx user:/nonexistent:/bin/false
user:x:1000:1000::/home/user:/bin/sh
user2:x:1001:1001::/home/user2:/bin/sh
```
On host, for example uid 1000 is me, anfedotoff:). So, If I start a process from `user (1000)` I will recieve my host user in event... What do you think about such case?

Reply:
Hm so:
1. You create a docker container that's running with uid 1001 right? since tetragon is reporting that. The uid is global in Tetragon view,  even if you map it with user namespaces, we always report uids in the host user namespace.
2. Then with the username lookup logic we will resolve it to host /etc/passwd, so inside host:/etc/passwd if uid 1001 is userA and inside container:/etc/passwd  uid 1001 is userB , in tetragon we will report userA , I guess that's the part you find confusing right?

I agree that users could be confused, let's then do it one step, we go with resolve only if process is in host NS, while we think about it, and we may just document it.

So to merge this, please keep your current change, just fix the test skipped, and we are good ;-)

Reply:
> Hm so:
> 
> 1. You create a docker container that's running with uid 1001 right? since tetragon is reporting that. The uid is global in Tetragon view,  even if you map it with user namespaces, we always report uids in the host user namespace.
> 2. Then with the username lookup logic we will resolve it to host /etc/passwd, so inside host:/etc/passwd if uid 1001 is userA and inside container:/etc/passwd  uid 1001 is userB , in tetragon we will report userA , I guess that's the part you find confusing right?

Yes, exactly. Let's resolve username only if process is in host NS for now!